clients/src/main/java/org/apache/kafka/clients/consumer/internals/AutoOffsetResetStrategy.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer.internals;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+
+import java.util.Objects;
+
+public class AutoOffsetResetStrategy {
+    public static final String EARLIEST_STRATEGY_NAME = "earliest";
+    public static final String LATEST_STRATEGY_NAME = "latest";
+    public static final String NONE_STRATEGY_NAME = "none";
+
+    public static final AutoOffsetResetStrategy EARLIEST = new AutoOffsetResetStrategy(EARLIEST_STRATEGY_NAME);
+    public static final AutoOffsetResetStrategy LATEST = new AutoOffsetResetStrategy(LATEST_STRATEGY_NAME);
+    public static final AutoOffsetResetStrategy NONE = new AutoOffsetResetStrategy(NONE_STRATEGY_NAME);
```

Comment:
If we use an `enum` we don't need this boiler plate objects?

Reply:
These are not boilerplate objects. These are static instances to represent current strategies.
The main reason for converting enum into class is to allow us to add additional reset strategies with some properties. 

For example, we will be adding new class like below to represent duration based reset  strategy: 
``` 
 DurationBasedOffsetResetStrategy extends AutoOffsetResetStrategy {
    DurationBasedOffsetResetStrategy(String name, String isoDuration) {
    }
 }    
```